src/main/java/seedu/duke/Ui.java

```diff
@@ -0,0 +1,128 @@
+package seedu.duke;
+
+import java.util.ArrayList;
+
+public class Ui {
+
+    private static boolean receivedBye = false;
+
+    /**
+     * Prints out a response, wrapped with 2 hyphenated lines above and below
+     * @param response
+     */
+    public static void print(String response) {
+        System.out.println("----------------------------------");
+        System.out.println(response.trim());
+        System.out.println("----------------------------------");
+    }
+
+    public static void showError(Exception e) {
+        System.out.println(e);
+    }
+
+    public static void showWelcome() {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);
+    }
+
+    public static void showBye() {
+        System.out.println("----------------------------------");
+        System.out.println("Bye. Hope to see you again soon!");
+        System.out.println("----------------------------------");
+    }
+
+    /**
+     * Responds to the user by printing a response, based on the chat sent by user
+     * @param chat User input
+     * @param ls The TaskList in Duke
+     */
+    public static void respond(String chat, TaskList ls) {
+        String keyword = chat.split(" ")[0];
+        switch(keyword) {
+        case("list"): {
+            ArrayList<Task> list = ls.getList();
+            String response = "";
+            for (int i = 0; i < list.size(); i++) {
+                Task task = list.get(i);
+                response += Integer.toString(i + 1) + ". ";
+                response += task.toString();
+                response += "\n";
+            }
+            Ui.print(response);
+            break;
+        }
+        case("mark"): {
+            Task task = ls.mark(chat.substring(5));
+            String response = "Nice! I've marked this task as done:\n";
+            response += task.toString();
+            Ui.print(response);
+            break;
+        }
+        case("unmark"): {
+            Task task = ls.unmark(chat.substring(7));
+            String response = "OK, I've marked this task as not done yet:\n";
+            response += task.toString();
+            Ui.print(response);
+            break;
+        }
```

Comment:
These lines have repeated code, perhaps they could be refactored?

```suggestion
        case("mark"): // Fallthrough
        case("unmark"): {
            boolean isDone = "mark".equals(keyword);
            String index = chat.substring(isDone ? 5 : 7);
            Task task = ls.mark(index, isDone);
            String response = "OK, I've marked this task as " + (isDone ? "done" : "not done") + ":\n";
            response += task.toString();
            Ui.print(response);
            break;
        }
```